src/lib.rs

```diff
@@ -401,14 +502,169 @@ impl Timer {
     pub fn schedule<F, T>(&self, date: DateTime<T>, repeat: Option<Duration>, cb: F) -> Guard
         where F: 'static + FnMut() + Send, T : chrono::offset::TimeZone
     {
-        let guard = Guard::new();
-        self.tx.send(Op::Schedule(Schedule {
-            date: date.with_timezone(&UTC),
-            cb: Box::new(cb),
-            guard: guard.clone(),
-            repeat: repeat
-        })).unwrap();
-        guard
+        self.base.schedule(date, repeat, Box::new(cb))
+    }
+}
+
+/// A timer, used to schedule delivery of messages at a later date.
+///
+/// In the current implementation, each timer is executed as two
+/// threads. The _Scheduler_ thread is in charge of maintaining the
+/// queue of messages to deliver and of actually deliverying them. The
+/// _Communication_ thread is in charge of communicating with the
+/// _Scheduler_ thread (which requires acquiring a possibly-long-held
+/// Mutex) without blocking the caller thread.
+pub struct MessageTimer<T>
+    where T : 'static + Send + Clone {
+    base: TimerBase<T>,
+    pub rx: Receiver<T>
```

Comment:
I like this, I might actually change MessageTimer::new() to take tx as a parameter, to allow mpsc such that the timer feeds into a Receiver that already exists